### DISCUSSION TITLE
How to pass span id to OpenAI `chat.completions.create`

### DISCUSSION THREAD

**Discussion by techytushar** [langfuse/langfuse#1091] [Support] [answered]
I can pass the `trace_id` to the OpenAi `chat.completions.create()` function, but what if I want to nest this Generation inside a span, how to do that ?
```
response = await openai_client.chat.completions.create(
        model=model,
        messages=messages,
        temperature=temperature,
        trace_id=trace_id,
        span_id=span_id,    # <--- this is not supported
    )
```
This is possible when using the `Generation`, `Span`, and `Trace` functions directly
```
trace = langfuse.trace(name = "llm-feature")
retrieval = trace.span(name = "retrieval")
retrieval.generation(name = "query-creation")
```

**Comment by marcklingen** [comment, marked as answer]
That's unfortunately currently unsupported but a great feature request. I've created an issue: #1092

**Reply by marcklingen** [reply]
@techytushar I added this in https://github.com/langfuse/langfuse-python/pull/381 and will release it later today
